src/core/MpiCallbacks.hpp

```diff
@@ -174,6 +198,28 @@ struct callback_one_rank_t final : public callback_concept_t {
   }
 };
 
+/**
+ * @brief Callback with a return value from the master rank.
+ *
+ * This is an implementation of a callback for a specific callable
+ * @p F and a set of arguments to call it with, where the value from
+ * the master rank is returned.
+ */
+template <class F, class... Args>
+struct callback_master_rank_t final : public callback_concept_t {
+  F m_f;
+
+  callback_master_rank_t(callback_master_rank_t const &) = delete;
+  callback_master_rank_t(callback_master_rank_t &&) = delete;
+
+  template <class FRef>
+  explicit callback_master_rank_t(FRef &&f) : m_f(std::forward<FRef>(f)) {}
+  void operator()(boost::mpi::communicator const &comm,
+                  boost::mpi::packed_iarchive &ia) const override {
+    auto const result = detail::invoke<F, Args...>(m_f, ia);
```

Comment:
I think the (unused) result should be explicitly casted to `void` to avoid unused variable warnings and make the intent clear. So

```c++
(void)detail::invoke<F, Args...>(m_f, ia);
```
or similar.

Reply:
Good idea!